### DISCUSSION TITLE
private repositori

### DISCUSSION THREAD

**Discussion by dzakyabdurhmn** [community/community#84942] [New to GitHub]
Hey, I was just casually checking out my GitHub api, but why did the commits of my private repo get leaked

this is endpoint:  https://api.github.com/users/user/events

```json
{
"id": "0000000",
"type": "CreateEvent",
"actor": {
"id": 0000000,
"login": "dzakyabdurhmn",
"display_login": "dzakyabdurhmn",
"gravatar_id": "",
"url": "https://api.github.com/users/dzakyabdurhmn",
"avatar_url": "https://avatars.githubusercontent.com/u/100127440?"
},
"repo": {
"id": 0000000,
"name": "dzakyabdurhmn/foo",
"url": "https://api.github.com/repos/dzakyabdurhmn/foo"
},
"payload": {
"ref": "main",
"ref_type": "branch",
"master_branch": "main",
"description": null,
"pusher_type": "user"
},
"public": true,
"created_at": "2024-01-06T17:04:53Z"
},
```

**Comment by bonizario** [comment]
The GitHub Events API only shows the events of public repositories by default. If you're seeing commits from a private repository, it's likely because you're authenticated and have the necessary permissions to view those events.

It could be a serious issue if you're not authenticated and still seeing private repository events. However, this is highly unlikely as GitHub has strict privacy controls.

If you're concerned about privacy, you can check your repository's settings to ensure it's set to private. Also, could you review the permissions of any personal access tokens you've created to make sure they don't have unnecessary access?

Remember, never share your personal access tokens, and avoid using them in client-side code or public repositories.

**Reply by dzakyabdurhmn** [reply]
Actually, it was originally a public repository, after a long time I changed it to private, but why are my previous commits still there in the API?

**Reply by bonizario** [reply]
When you change a repository from public to private, the commits made while the repository was public will still be visible in the GitHub API, but only to users with the necessary permissions to access the repository.

The GitHub API respects the same visibility and access controls as the GitHub website. This means that if a repository is private, its data (including commits) can only be accessed via the API by authenticated users with permission.

If you want to remove a commit from the repository's history (and therefore from the GitHub API), you need to use Git's history rewriting commands, such as `git rebase` or `git filter-branch.` However, I want you to know that it can be disruptive to other users and should be done carefully.

**Reply by dzakyabdurhmn** [reply]
OK, thank you very much for your help

**Reply by bonizario** [reply]
No problem! Could you please mark this answer as the "accepted"?

If you have any other questions, feel free to ask 🚀
